src/tests/QueryExecutor.test.ts

```diff
@@ -0,0 +1,316 @@
+      // set to any to access private properties
+      const queryExecutor = new QueryExecutor("QueryExecutor", config) as any;
+      expect(queryExecutor.config).toBe(config);
+    });
+
+    it("should call loadSchema", () => {
+      new QueryExecutor("QueryExecutor", config) as any;
+      expect(loadSchemaSpy).toHaveBeenCalled();
+    });
+
+    it("should call findQueries", () => {
+      // set to any to access private properties
+      const findQueriesSpy = jest.spyOn(
+        QueryExecutor.prototype as any,
+        "findQueries"
+      );
+      new QueryExecutor("QueryExecutor", config) as any;
+      expect(findQueriesSpy).toHaveBeenCalled();
+      findQueriesSpy.mockRestore();
+    });
+  });
+
+  describe("execQueries", () => {
+    let executeQuerySpy: jest.SpyInstance;
+    let handleQueryResultsSpy: jest.SpyInstance;
+    let loadSchemaSpy: jest.SpyInstance;
+    let queryExecutor: QueryExecutor;
+
+    beforeAll(() => {
+      // set to any to access private properties
+      executeQuerySpy = jest
+        .spyOn(QueryExecutor.prototype as any, "executeQuery")
+        .mockResolvedValue({ data: [] });
+      handleQueryResultsSpy = jest
+        .spyOn(QueryExecutor.prototype as any, "handleQueryResults")
+        .mockResolvedValue({});
+      loadSchemaSpy = jest
+        .spyOn(QueryExecutor.prototype as any, "loadSchema")
+        .mockResolvedValue(true);
+
+      queryExecutor = new QueryExecutor("test", config);
+      // set to access private properties
+      // @ts-ignore
+      queryExecutor.queries2Exec = ["test"];
+    });
+
+    afterEach(() => {
+      executeQuerySpy.mockReset().mockResolvedValue({ data: [] });
+      handleQueryResultsSpy.mockReset().mockResolvedValue({});
+    });
+
+    afterAll(() => {
+      executeQuerySpy.mockRestore();
+      handleQueryResultsSpy.mockRestore();
+      loadSchemaSpy.mockRestore();
+    });
+
+    it("should add lastRun", async () => {
+      await queryExecutor.execQueries();
+
+      expect(executeQuerySpy).toHaveBeenCalledTimes(1);
+      const variables = executeQuerySpy.mock.calls[0][1] as IGraphqlVariables;
+      // set to any to access private properties
+      expect(variables.lastRun).toBe((queryExecutor as any).lastSuccessfulRun);
+    });
+
+    it("should keep variables", async () => {
+      await queryExecutor.execQueries({ blocknumber: 123 });
+
+      expect(executeQuerySpy).toHaveBeenCalledTimes(1);
+      const variables = executeQuerySpy.mock.calls[0][1] as IGraphqlVariables;
+      // set to any to access private properties
+      expect(variables.lastRun).toBe((queryExecutor as any).lastSuccessfulRun);
+      expect(variables.blocknumber).toBe(123);
+    });
+
+    it("should log error", async () => {
+      executeQuerySpy.mockRejectedValueOnce(new Error("test"));
+      await queryExecutor.execQueries();
+
+      // set to any to access private properties
+      expect((queryExecutor as any).logger.error).toHaveBeenCalled();
```

Comment:
I could but then I need to resort to this pattern because testing a log with a stack trace in it is kinda hard:
```javascript
expect((queryExecutor as any).logger.error).toHaveBeenCalledTimes(1);
expect((queryExecutor as any).logger.error.mock.calls[0][0]).toHaveBeenCalledWith('Failed executing test query with');
```